pkg/meta/model/job_args.go

```diff
@@ -1331,6 +1331,12 @@ type IndexArg struct {
 	// But we keep the json field name as `is_vector` for compatibility.
 	IsColumnar bool `json:"is_vector,omitempty"`
 
+	// ColumnarIndexType is used to distinguish different columnar index types.
+	// Only used for job args v2.
+	// Note: 1. when you want to read it, always calling `GetColumnarIndexType`` rather than using it directly.
+	//       2. when you set it, make sure IsColumnar = ColumnarIndexType != ColumnarIndexTypeNA.
```

Comment:
Then how about provide a `SetColumnarIndexType` to handle `IsColumnar` together?